src/classes/STG_InstallScript.cls

```diff
@@ -284,4 +237,69 @@ global without sharing class STG_InstallScript implements InstallHandler {
             upsert orgSettings;
         }
     }
+
+    /*******************************************************************************************************
+    * @description Schedules a class that implements the Schedulable interface, if it's not already scheduled.
+    * @param JobName The name to give the scheduled job.
+    * @param frequency The frequency to schedule the job at.
+    * @param className The class to schedule. Should implement the Schedulable interface.
+    * @return The ID of the scheduled job.
+    */
+    public static String scheduleJobIfNotScheduled(String JobName, String frequency, String className) {
+
+        //Check if the job is already scheduled - CronJobDetail is the parent and CronTrigger is the child
+        //Type '7' is for scheduled Apex
+        List<CronTrigger> scheduledJobs = [select Id, CronJobDetail.Id from CronTrigger
+                    where CronJobDetail.Name = :JobName and CronJobDetail.JobType = '7'];
+
+        if(scheduledJobs.size() == 0) {
+            Type classType = Type.forName(className);
+
+            if(classType != null) {
+               Object instance = classType.newInstance();
+               if(instance instanceof Schedulable)
+                   return System.schedule(JobName, frequency, (Schedulable)instance);
+            }
+        }
+        return null;
+    }
+
+    /*******************************************************************************************************
+    * @description: Populate the Owned by Namespace field for EDA owned triggers. We have to update Trigger
+    * handlers even if they are marked as user managed.
+    */
+    private static void populateNamespaceHandlers() {
+        // Populate filters for soql query
+        Set<String> objectTypes = new Set<String>();
+        Set<String> classNames = new Set<String>();
+
+        for(TDTM_Global_API.TdtmToken token : TDTM_Global_API.getDefaultTdtmConfig() ) {
+            objectTypes.add(token.targetObject);
+            classNames.add(token.className);
+        }
+
+        // Get EDA trigger Handlers from the database
+        List<Trigger_Handler__c> hedaHandlers = [SELECT Id, Name, Owned_by_Namespace__c FROM Trigger_Handler__c WHERE Class__c in: classNames AND Object__c in: objectTypes LIMIT 5000];
+
+        for(Trigger_Handler__c t : hedaHandlers) {
```

Comment:
Although they're hard to read, should we consider using the SOQL for loop here? I see we are retrieving 5000 records as a limit (though I'd be very shocked if we had anyone even that close). However if someone were to have 5000 records actually returned, the loop would run 5000 times versus using the SOQL for loop with a list which would run that for loop in chunks of 200 which would run the loop 25 times. (Granted the loop isn't doing much). Just thinking "out loud" :)

Reply:
It looks like this only runs when upgrading from a version lower than 1.38, so I don't think it's worth the effort to change. (I could update the ApexDocs to reflect that though, as it's probably something we can remove eventually...)